src/main/java/vazkii/botania/common/item/rod/ItemTornadoRod.java

```diff
@@ -146,20 +151,45 @@ public void onAvatarUpdate(IAvatarTile tile, ItemStack stack) {
 			int rangeY = 3;
 			List<PlayerEntity> players = world.getEntitiesWithinAABB(PlayerEntity.class, new AxisAlignedBB(te.getPos().add(-0.5 - range, -0.5 - rangeY, -0.5 - range), te.getPos().add(0.5 + range, 0.5 + rangeY, 0.5 + range)));
 			for (PlayerEntity p : players) {
-				if (p.getMotion().getY() > 0.3 && p.getMotion().getY() < 2 && !p.isSneaking()) {
-					p.setMotion(p.getMotion().getX(), 2.8, p.getMotion().getZ());
+				if (!p.isSneaking()) {
+					if (p.getMotion().length() > 0.2 && p.getMotion().length() < 1.20 && p.isElytraFlying()) {
+						Vector3d lookDir = (p.getLookVec());
+						p.setMotion(p.getMotion().getX() + lookDir.getX() * 1.25
+								, p.getMotion().getY() + lookDir.getY() * 1.25
+								, p.getMotion().getZ() + lookDir.getZ() * 1.25);
+						for (int i = 0; i < 20; i++) {
+							for (int j = 0; j < 5; j++) {
+								WispParticleData data = WispParticleData.wisp(0.35F + (float) Math.random() * 0.1F, 0.25F, 0.25F, 0.25F);
+								world.addParticle(data, p.getPosX() + lookDir.getX() * i
+										, p.getPosY() + lookDir.getY() * i
+										, p.getPosZ() + lookDir.getZ() * i
+										, 0.2F * (float) (Math.random() - 0.5) * (Math.abs(lookDir.getY()) + Math.abs(lookDir.getZ())) + -0.01F * (float) Math.random() * lookDir.getX()
+										, 0.2F * (float) (Math.random() - 0.5) * (Math.abs(lookDir.getX()) + Math.abs(lookDir.getZ())) + -0.01F * (float) Math.random() * lookDir.getY()					//FIX YOUR SHIT, MAKE PARTICLES GO THE RIGHT WAY
+										, 0.2F * (float) (Math.random() - 0.5) * (Math.abs(lookDir.getY()) + Math.abs(lookDir.getX())) + -0.01F * (float) Math.random() * lookDir.getZ());
+							}
+						}
```

Comment:
pull the elytra part out into a separate method for cleanliness (you can pull the non elytra side out too if you want, but you don't have to)

Reply:
Oh whoops, I see I left in some stupid comments-
Yeah I'll look into doing that when I get the time

Reply:
So did some cleanup, and changed the way the boosts are implemented to allow for stacked boosts to be a lot more smooth. Did this by increasing the hardcap, adding a short cooldown, and applying diminishing returns based on current speed.
-Though, actual values might need some tweaking, but that's something I imagine would require more actual survival play.